executor/adapter.go

```diff
@@ -1370,6 +1375,11 @@ func (a *ExecStmt) FinishExecuteStmt(txnTS uint64, err error, hasMoreResults boo
 		// but we set it again in case we missed some code paths.
 		sessVars.StmtCtx.SetPlan(a.Plan)
 	}
+
+	/// Triage for historical stats
+	if succ {
+		a.RecordHistoryStats()
```

Comment:
I think it won't. Because the record work will be done only when estimation is not accurate. For benchmarks, when "warm up", these estimation should be adjusted if "optimizer_history_stats" is on.

Reply:
I agree, when the write path is completed, I'll do some benchmark testing.